### DISCUSSION TITLE
User profile after login

### DISCUSSION THREAD

**Discussion by bat0ri** [fastapi-users/fastapi-users#1298] [Q&A]
Good afternoon everyone, I just started learning web technologies.

- I'm making a front on React, and I want to display at least some information about the user

```javascript
const ProfileComponent = () => {
 const [profileData, setProfileData] = useState(null);
  useEffect(() => {
    const getCookie = (name) => {
      const value = `; ${document.cookie}`;
      const parts = value.split(`; ${name}=`);
      if (parts.length === 2) return parts.pop().split(';').shift();
    };  

   const authToken = getCookie('AUTH');
    if (!authToken) {
      console.error('JWT token not found in Cookie.');
      return;
    }
    const headers = {
      Authorization: `Bearer ${authToken}`,
    };
    axios
      .get('http://127.0.0.1:8000/chat/chat', { headers })
      .then((response) => {
        setProfileData(response.data);
      })
      .catch((error) => {
        console.error('Error fetching profile data:', error);
      });
  }, []);
``` 
 

console log -------->>>> JWT token not found in Cookie.

- I still don’t understand how to display user information

```python
@chat.get("/chat")
def protected_route(user: User = Depends(current_user)):
    return user.email
``` 

- one more question

for login I used this react component, how right I did?????????
```javascript
const navigate = useNavigate();
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
  

    const handleSubmit = async (e) => {
    e.preventDefault();

    const formData = new URLSearchParams();
    formData.append('grant_type', 'password');
    formData.append('username', username);
    formData.append('password', password);
    formData.append('scope', '');
    formData.append('client_id', '');
    formData.append('client_secret', '');

    try {
        const response = await axios.post('http://localhost:8000/auth/login', formData, {
        headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
        },
            withCredentials: true, 
        });

        if (response.status === 204) {
            console.log('Вход выполнен успешно');
            navigate('/chat');
        } else {
            console.error('Ошибка входа');
        }
    } catch (error) {
        console.error('Ошибка при отправке запроса:', error);
    }
    };
```

**Comment by hgalytoby** [comment]
> Good afternoon everyone, I just started learning web technologies.
> 
> * I'm making a front on React, and I want to display at least some information about the user
> 
> ```js
> const ProfileComponent = () => {
>  const [profileData, setProfileData] = useState(null);
>   useEffect(() => {
>     const getCookie = (name) => {
>       const value = `; ${document.cookie}`;
>       const parts = value.split(`; ${name}=`);
>       if (parts.length === 2) return parts.pop().split(';').shift();
>     };  
> 
>    const authToken = getCookie('AUTH');
>     if (!authToken) {
>       console.error('JWT token not found in Cookie.');
>       return;
>     }
>     const headers = {
>       Authorization: `Bearer ${authToken}`,
>     };
>     axios
>       .get('http://127.0.0.1:8000/chat/chat', { headers })
>       .then((response) => {
>         setProfileData(response.data);
>       })
>       .catch((error) => {
>         console.error('Error fetching profile data:', error);
>       });
>   }, []);
> ```
> 
> console log -------->>>> JWT token not found in Cookie.
> 
> * I still don’t understand how to display user information
> 
> ```python
> @chat.get("/chat")
> def protected_route(user: User = Depends(current_user)):
>     return user.email
> ```
> 
> * one more question
> 
> for login I used this react component, how right I did?????????
> 
> ```js
> const navigate = useNavigate();
>     const [username, setUsername] = useState('');
>     const [password, setPassword] = useState('');
>   
> 
>     const handleSubmit = async (e) => {
>     e.preventDefault();
> 
>     const formData = new URLSearchParams();
>     formData.append('grant_type', 'password');
>     formData.append('username', username);
>     formData.append('password', password);
>     formData.append('scope', '');
>     formData.append('client_id', '');
>     formData.append('client_secret', '');
> 
>     try {
>         const response = await axios.post('http://localhost:8000/auth/login', formData, {
>         headers: {
>             'Content-Type': 'application/x-www-form-urlencoded',
>         },
>             withCredentials: true, 
>         });
> 
>         if (response.status === 204) {
>             console.log('Вход выполнен успешно');
>             navigate('/chat');
>         } else {
>             console.error('Ошибка входа');
>         }
>     } catch (error) {
>         console.error('Ошибка при отправке запроса:', error);
>     }
>     };
> ```

I see you've written `const authToken = getCookie('AUTH')`, so when you log in, you should put the login response's access_token into the 'AUTH' cookie.
